test/controllers/processors/CompanyNameProcessor.test.ts

```diff
@@ -0,0 +1,133 @@
+
+describe('CompanyNameProcessor', () => {
+
+    it('should throw an error if session is not found', async () => {
+
+        const companyNameProcessor = new CompanyNameProcessor(Substitute.for<CompaniesHouseSDK>());
+        const request = {} as Request;
+
+        try {
+            await companyNameProcessor.process(request);
+            assert.fail('Expected to throw error');
+        } catch (err) {
+            expect(err.message).to.equal(SESSION_NOT_FOUND_ERROR.message);
+        }
+
+    });
+    it('should throw an error if access token is missing from session', async () => {
+
+        const companyNameProcessor = new CompanyNameProcessor(Substitute.for<CompaniesHouseSDK>());
+        const session = createSession('secret', true);
+        delete session.data.signin_info?.access_token?.access_token;
+
+        const request = {
+            session
+        } as Request;
+
+        try {
+            await companyNameProcessor.process(request);
+            assert.fail('Expected to throw error');
+        } catch (err) {
+            expect(err.message).to.equal(TOKEN_MISSING_ERROR.message);
+        }
+
+    });
+    it('should throw an error if company number is not in the request body', async () => {
+
+        const companyNameProcessor = new CompanyNameProcessor(createSubstituteOf<CompaniesHouseSDK>());
+        const session = createSession('secret');
+
+        const request = {
+            session,
+            body: {
+            }
+        } as Request;
+
+        try {
+            await companyNameProcessor.process(request);
+            assert.fail('Expected to throw error');
+        } catch (err) {
+            expect(err.message).to.equal(COMPANY_NUMBER_UNDEFINED_ERROR.message);
+        }
+    });
+    it('should throw an error if sdk fails to retrieve company profile', async () => {
+
+        const companyProfileService = createSubstituteOf<CompanyProfileService>();
+
+        companyProfileService.getCompanyProfile(Arg.any()).resolves({
+            httpStatusCode: 404
+        });
+
+        const companiesHouseSDK = (_: AuthMethod) => createSubstituteOf<ApiClient>(sdk => {
+            sdk.companyProfile.returns!(companyProfileService);
+        });
+        const companyNameProcessor = new CompanyNameProcessor(companiesHouseSDK);
+        const session = createSession('secret');
+
+        const companyNumber = 'NI000000';
+
+        const request = {
+            session,
+            body: {
+                companyNumber
+            }
+        } as Request;
+
+        try {
+            await companyNameProcessor.process(request);
+            assert.fail('Expected to throw error');
+        } catch (err) {
+            companyProfileService.received().getCompanyProfile(companyNumber);
+            expect(err.message).to.equal(COMPANY_NAME_RETRIEVAL_ERROR(companyNumber).message);
+        }
+    });
+    it('should add the company name to the penaltyIdentifier in the session', async () => {
```

Comment:
how are we verifying the company name is added to the session here?

Reply:
It's implicitly tested by BaseController tests. I'm only testing the functionality of this code.

Reply:
could just rename the test to reflect successful retrieval of company name or add a check to confirm the data has been added to body